src/never.rs

```diff
@@ -0,0 +1,15 @@
+use std::{error::Error, fmt};
+
+/// A type representing a value that can never materialize.
+///
+/// This would be `!`, but it isn't stable yet.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Never {}
```

Comment:
TIOLI: Maybe we ought to put this in a crate in `lib` so that `lib/stack` and the main binary can use the same definition of the never type? May not be worth it...

Reply:
that seems like a good idea to me